sepal_ui/sepalwidgets/app.py

```diff
@@ -380,3 +409,122 @@ def add_banner(self, msg, **kwargs):
+
+    def _locale_info(self, change):
+        """display information about the locale change"""
+
+        if change["new"] != "":
+            msg = ms.locale.change.format(change["new"])
+            self.add_banner(msg, type="success")
+
+        return
+
+
+class LocaleSelect(v.Menu, SepalWidget):
+    """
+    An language selector for sepal-ui based application.
+
+    It displays the currently requested language (not the one used by the translator).
+    When value is changed, the sepal-ui config file is updated. It is designed to be used in a AppBar component.
+
+    .. warning:: as the component is a v.Menu to get the selected value you need to lisen to "value" instead of "v_model".
+
+    .. versionadded:: 2.7.0
+
+
+    Args:
+        translator (sw.Translator, optional): the translator of the app, to match the used language
+        kwargs (dict, optional): any arguments for a Btn object, children will be override
+    """
+
+    COUNTRIES = pd.read_csv(Path(__file__).parents[1] / "scripts" / "locale.csv")
+    "pandas.DataFrame: the country list as a df. columns [code, name, flag]"
+
+    FLAG = "https://flagcdn.com/{}.svg"
+    "str: the url of the svg flag images"
+
+    ATTR = {"src": "https://flagcdn.com/gb.svg", "width": "30", "alt": "en-UK"}
+    "dict: the default flag parameter, default to english"
+
+    btn = None
+    "v.Btn: the btn to click when changing language"
+
+    language_list = None
+    "v.List: the list of countries with their flag,name in english, and ISO code"
+
+    def __init__(self, translator=None, **kwargs):
+
+        # extract the language information from the translator
+        # if not set default to english
+        code = "en-US" if translator is None else translator.target
+        loc = self.COUNTRIES[self.COUNTRIES.code == code].squeeze()
+        attr = {**self.ATTR, "src": self.FLAG.format(loc.flag), "alt": loc.name}
+
+        kwargs["small"] = kwargs.pop("small", True)
+        kwargs["v_model"] = False
+        kwargs["v_on"] = "x.on"
+        kwargs["children"] = [v.Html(tag="img", attributes=attr, class_="mr-1"), code]
+        self.btn = v.Btn(**kwargs)
+
+        self.language_list = v.List(
+            dense=True,
+            flat=True,
+            color="grey darken-3",
+            v_model=True,
+            max_height="300px",
+            style_="overflow: auto; border-radius: 0 0 0 0;",
+            children=[v.ListItemGroup(children=self._get_country_items(), v_model="")],
+        )
+
+        super().__init__(
+            children=[self.language_list],
+            v_model=False,
+            close_on_content_click=True,
+            v_slots=[{"name": "activator", "variable": "x", "children": self.btn}],
+            value=loc.code,
+        )
+
+        # add js behaviour
+        jsdlink((self.language_list.children[0], "v_model"), (self, "value"))
+        self.language_list.children[0].observe(self._on_locale_select, "v_model")
+
+    def _get_country_items(self):
```

Comment:
What do you think if we only get the available languages instead of listing all of them? it won't be useful for the users if they click and the language is not available.

Reply:
That was my initial thought but as it's a general parameter for all apps I decided it would be easier to make it always general. My idea is: it will only be set once in any app and then you rarely change the parameter again. If you have a different set of languages depending on the app, that can be disturbing e.g.:   
- your default language is "Albanian"
- App does not support it. You want to swith to another one but you are still not satisfied.
- then to switch back to "Albanian" you need to find in SEPAL the one and only app that support this language.

I agree though that the user should know which languages are supported in the app he's using. 
There are 3 informations to display: 
- what language is supported
- what is the current default
- what is the currently used one

maybe adding some chips in the list next to the flags ?